tools/slang-unit-test/unit-test-coverage-tracing-metadata.cpp

```
@@ -131,22 +131,36 @@ SLANG_UNIT_TEST(coverageTracingMetadata)
-    // string and line number must both be populated for every slot;
-    // the synthesizer gives every counter op a real source location.
-    for (uint32_t i = 0; i < counterCount; ++i)
+    uint32_t entryCount = coverage->getEntryCount();
+    SLANG_CHECK(entryCount == counterCount);
+
+    // Walk every source entry and exercise the per-entry accessor.
+    // File string and line number must both be populated for every
+    // current line-coverage entry; the synthesizer gives every
+    // counter op a real source location.
+    for (uint32_t i = 0; i < entryCount; ++i)
     {
```

Cppcheck (2.20.0)

[portability] Shifting signed 32-bit value by 31 bits is implementation-defined behaviour

(shiftTooManyBitsSigned)